### PR TITLE
Provide speech symbols file name path on creation

### LINKED ISSUE #17344: nvda will fail to close the pronounciation/symbol pronounciation dialog when the ok button is envoked.
### Steps to reproduce:
1, cleanly install NVDA. uninstall NVDA completly from programs and features, and delete your user config in percent appdata percent to completely start over from scratch with nvda. 2. reinstall nvda in the normal way, preferably 2024.4 or 2024.4 rc 1 will do as well.
3, when NVDA is installed, go to the nvda menu with insert or capslock n then p for preferences. then go to the Punctuation/symbol pronunciation dialog.
4, once their, immidiately press the ok button, you will notice NVDA will not close the dialog and save your changes. the button will say pressed, but the dialog will stay open.
### Actual behavior:

NVDA+n
NVDA  menu
Settings...  s
down arrow
Speech dictionaries  subMenu  d
down arrow
Punctuation/symbol pronunciation...  p
enter
Symbol Pronunciation (English)  dialog
Symbols  list  Alt+  s
. sentence ending; Replacement: dot; Level: all; Preserve: always  1 of 4579
tab
Change selected symbol  grouping
Replacement  edit  Alt+  r  selected dot
tab
Level  combo box  all  collapsed  Alt+  l
tab
Send actual symbol to synthesizer  combo box  always  collapsed  Alt+  s
tab
Add  button  Alt+  a
tab
OK  button
pressed
pressed
tab
Cancel  button
pressed

### Expected behavior:

NVDA+n
NVDA  menu
p
Settings...  s
down arrow
Speech dictionaries  subMenu  d
down arrow
Punctuation/symbol pronunciation...  p
enter
Symbol Pronunciation (English)  dialog
Symbols  list  Alt+  s
. sentence ending; Replacement: dot; Level: all; Preserve: always  1 of 4579
tab
Change selected symbol  grouping
Replacement  edit  Alt+  r  selected dot
tab
Level  combo box  all  collapsed  Alt+  l
tab
Send actual symbol to synthesizer  combo box  always  collapsed  Alt+  s
tab
Add  button  Alt+  a
tab
OK  button
space
pressed
NVDA Speech Viewer
Show Speech Viewer on Startup  check box  not checked  Alt+  s
NVDA+4
speak command keys off

### NVDA logs, crash dumps and other attachments:
NVDA debug log is attached.
#### NVDA installed/portable/running from source:
NVDA Installed
#### NVDA version:
2024.4 rc1/2024.4
#### Windows version:
windows 11 24H2
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes. older versions of NVDa from 2024.3 and older do not Have this behavior.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
[nvda-old wensday october 30, 2024.log](https://github.com/user-attachments/files/17581322/nvda-old.wensday.october.30.2024.log)

### PR DESCRIPTION
**Pull request by seanbudd**
### Link to issue number:
Fixes #17344

### Summary of the issue:

The `SpeechSymbols` only sets the filename to write to when the dictionary is loaded.
Before 826ef91, dictionary files were always loaded before they were saved.
Afterwards, dictionaries are saved without being loaded.
When trying to save dictionary files the following error occurs, due to no file name being set

```py
Speaking [LangChangeCommand ('en_US'), 'pressed']
ERROR - unhandled exception (17:00:20.842) - MainThread (9964):
Traceback (most recent call last):
  File "gui\settingsDialogs.pyc", line 5350, in onOk
  File "characterProcessing.pyc", line 364, in save
ValueError: No file name
```

### Description of user facing changes

Users can now save the speech dictionary dialog.
Add-ons which use custom speech dictionaries are also saved correctly now.

### Description of development approach

Create optional parameter to set filename path for speech dictionaries.

### Testing strategy:

Manual testing issue described in #17345
Manually testing Perky Duck add-on which uses a custom speech dictionary

### Known issues with pull request:

None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Release Notes

- **New Features**
	- Introduced support for new braille displays with automatic detection.
	- Enhanced support for Microsoft Office applications, including better formatting and table reporting.
	- Added functionality for reading and interacting with mathematical content in web browsers and Office.
	- Implemented automatic language switching based on document language.
	- Launched a new "Add-on Store" for managing NVDA add-ons.

- **Improvements**
	- Significant enhancements to web browsing support, including a new "browse mode."
	- Improved stability and performance with numerous bug fixes.

- **Version Update**
	- Minor version updated from 0 to 1.

### CODE REVIEW

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # What's New in NVDA
 
+## 2024.4.1
+
+This patch release includes a fix for Outlook on certain Windows 10 systems.
+It also includes a fix for saving speech symbol dictionaries.
+
+### Bug fixes
+
+* Fixed bug where speech dictionaries were not saved and the dialog would not be closed. (#17345)
```
Comment (seanbudd) [inline user_docs/en/changes.md:10]:
```suggestion
* Fixed bug where speech dictionaries were not saved and the dialog would not be closed. (#17344)
```

**Review by LeonarddeR — APPROVED** [review]
I definitely prefer this approach, thanks!

**Comment by nvdaes** [comment]
For me this sounds good, but I'm not able to test this. I get the following error:

 CRITICAL - __main__ (09:37:46.570) - MainThread (6652):
core failure
Traceback (most recent call last):
  File "nvda.pyw", line 526, in <module>
    core.main()
  File "core.py", line 843, in main
    import UIAHandler
  File "UIAHandler\__init__.py", line 30, in <module>
    import IAccessibleHandler.internalWinEventHandler
  File "IAccessibleHandler\__init__.py", line 51, in <module>
    import NVDAObjects.IAccessible
  File "NVDAObjects\IAccessible\__init__.py", line 31, in <module>
    from comInterfaces.tom import ITextDocument
  File "comInterfaces\tom.py", line 3, in <module>
    import comtypes.gen._8CC497C9_A1DF_11CE_8098_00AA0047BE5D_0_1_0 as __wrapper_module__
  File "comInterfaces\_8CC497C9_A1DF_11CE_8098_00AA0047BE5D_0_1_0.py", line 2816, in <module>
    _check_version('1.4.6', 1728029317.614646)
  File "monkeyPatches\comtypesMonkeyPatches.py", line 197, in _check_version
    raise ImportError("Wrong version")
ImportError: Wrong version

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -1,5 +1,14 @@
 # What's New in NVDA
 
+## 2024.4.1
+
+This patch release includes a fix for Outlook on certain Windows 10 systems.
+It also includes a fix for saving speech symbol dictionaries.
+
+### Bug fixes
+
+* Fixed bug where speech dictionaries were not saved and the dialog would not be closed. (#17344)
```
Comment (nvdaes) [inline user_docs/en/changes.md:10]:
```suggestion
* Fixed bug where speech symbols dictionaries were not saved and the dialog would not be closed. (#17344)
```

**Comment by nvdaes** [comment]
Sorry, I didn't mereged this into rc but into master. I'm recreating the virtual environment on my machine to test properly.

**Comment by nvdaes** [comment]
I'll test the artifact when appveyord creates it.

**Comment by nvdaes** [comment]
Also, in `source/addonAPIVersion.py`, I think we should made this change:

- BACK_COMPAT_TO: AddonApiVersionT = (2025, 1, 0)
+ BACK_COMPAT_TO: AddonApiVersionT = (2024, 1, 0)

2024.4.1 doesn't exist in the server as an API version, so add-ons cannot be retrievet when trying the artifact.

**Review by nvdaes — APPROVED** [review]

**Comment by kolohekai** [comment]
lets hope this issue is resolved. looking forward to testing the alpha snapshot!!!

File: user_docs/en/changes.md [hunk 3]
```diff
@@ -1,5 +1,14 @@
 # What's New in NVDA
 
+## 2024.4.1
+
+This patch release includes a fix for Outlook on certain Windows 10 systems.
```
Comment (SaschaCowley) [inline user_docs/en/changes.md:5]: This should not be mentioned until we actually have a fix.

**Review by SaschaCowley — APPROVED** [review]

File: user_docs/en/changes.md (hunk 3 above)
Comment (seanbudd) [reply user_docs/en/changes.md:5]:
```suggestion
```

**Comment by seanbudd** [comment]
@nvdaes - no, we just need to update `nvdaAPIVersions.json`. This is a 2024.4.1 release, we are not using the 2025.1 API

**Pull request merged**